tensorflow/compiler/mlir/lite/transforms/legalize_tf.cc

```diff
@@ -641,6 +645,17 @@ void LegalizeTF::runOnFunction() {
   auto* context = &getContext();
   auto func = getFunction();
 
+  TFL::PopulatePrepareTfPatterns(ctx, &patterns);
```

Comment:
Instead of directly using all the patterns here, we should move the relevant patterns out of the prepare tf pass. Patterns that are generic can go to lower_tf pass and rest of the TFLite specific patterns can be in legalize_tf pass.

However, it is better to do that in two separate PR follow-up PRs to this one. One to move all the relevant patterns here and then to move general patterns to lower_tf pass. We should just introduce a TODO here.